### PR TITLE
Jetpack Manage: Making sure filter selections are cleared or corrected when choosing submenu items

### PR DESCRIPTION
**Pull request by coder-karen**
Related to https://github.com/Automattic/jetpack-manage/issues/322

## Proposed Changes

* This PR allows a prop - 'path' - to be passed down into `DashboardOverview` and then into `SitesDashboardV2` via `SitesOverviewContext.Provider`. This path prop contains the correct path whether a filter is selected in the Sites V2 dashboard, or a sites submenu item clicked. This then allows us to check if the path prop has changed, and then either clear the filter or correct the filter if needed.
* In practice, this means that on clicking the 'Favorites' or 'Needs Attention' menu items, the filter will be cleared. If adding a filter on the Favorites page, that filter will work with the selected favorites as it's base. And if clicking on the 'Needs Attention' link, the filter will change to 'Needs Attention'.

## Testing Instructions

* To replicate the original issue, on trunk navigate to `http://jetpack.cloud.localhost:3000/sites`. Select a filter from the filter dropdown. Then, navigate to any of the sub-menu item ('Favorites', 'Needs Attention' or 'All'). Notice that the filter is still showing in the UI (even though not active). 
* To test the fix, apply this PR and follow the same steps. The filter will clear on 'Favorites' and 'All', and 'Needs Attention' will correctly show the 'Needs Attention' filter.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by cleacos** [comment]
❤️  I tested it, and the solution works perfectly. 

Could we do the same behavior for the search box too? If I click on one of the preset filters on the left-side submenu with an active search, the list is updated with the filter, but the search textbox still has content; it should be empty. We are resetting any active filter or search.

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx [hunk 1]
```diff
@@ -129,6 +129,19 @@ export default function SitesDashboardV2() {
...
 
+	// Set or clear filter depending on sites submenu path selected
+	useEffect( () => {
+		if ( path === '/sites' || path === '/sites/favorites' ) {
+			setSitesViewState( { ...sitesViewState, filters: [] } );
+		}
+		if ( path === '/sites?issue_types=all_issues' ) {
+			setSitesViewState( {
+				...sitesViewState,
+				filters: [ { field: 'status', operator: 'in', value: 1 } ],
```
Comment (cleacos) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx:140]: Maybe:

```suggestion
				filters: [ { field: 'status', operator: 'in', value: 1 } ],
				search: '',
```

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx [hunk 2]
```diff
@@ -129,6 +129,19 @@ export default function SitesDashboardV2() {
 		updateDashboardURLQueryArgs( { search: sitesViewState.search } );
 	}, [ sitesViewState.search ] );
 
+	// Set or clear filter depending on sites submenu path selected
+	useEffect( () => {
+		if ( path === '/sites' || path === '/sites/favorites' ) {
+			setSitesViewState( { ...sitesViewState, filters: [] } );
```
Comment (cleacos) [inline client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx]:
```suggestion
			setSitesViewState( { ...sitesViewState, filters: [], search: '' } );
```

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx (hunk 1 above)
Comment (coder-karen) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx:140]: ❤️

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx (hunk 2 above)
Comment (coder-karen) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx]: ❤️

**Comment by coder-karen** [comment]
> Could we do the same behavior for the search box too? If I click on one of the preset filters on the left-side submenu with an active search, the list is updated with the filter, but the search textbox still has content; it should be empty. We are resetting any active filter or search.

That's a good idea, and those suggestions make sense to me.

**Review by cleacos — APPROVED** [review]
LGTM!

File: client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx (hunk 1 above)
Comment (tbradsha) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx:140]: Note that the value of item 1 might change, so this is a bit of a magic number... 😅

Also, if "Needs attention" had its own route, there'd be no need for a special condition here.

Comment (cleacos) [reply client/jetpack-cloud/sections/agency-dashboard/sites-overview/sites-dashboard-v2.tsx:140]: Yep, that is inherited from DataViews; they use magic numbers for referencing to filters.

**Pull request merged**